src/row_codec.rs

```diff
@@ -1,182 +1,230 @@
-use crate::db_state::RowFeature;
 use crate::error::SlateDBError;
-use crate::types::{KeyValueDeletable, RowAttributes, ValueDeletable};
+use crate::types::{RowEntry, ValueDeletable};
 use bitflags::bitflags;
 use bytes::{Buf, BufMut, Bytes, BytesMut};
 
 bitflags! {
+    #[derive(Debug, Clone, PartialEq, Default)]
     pub(crate) struct RowFlags: u8 {
         const Tombstone = 0b00000001;
+        const HAS_EXPIRE_TS = 0b00000010;
+        const HAS_CREATE_TS = 0b00000100;
     }
 }
 
-const NO_EXPIRE_TS: i64 = i64::MIN;
-
 /// Encodes key and value using the binary codec for SlateDB row representation
-/// using the `v0` encoding scheme.
+/// using the `v1` encoding scheme.
 ///
-/// The `v0` codec for the key is (for non-tombstones):
+/// The `v1` codec for the key is (for non-tombstones):
 ///
 /// ```txt
-///  |--------------------------------------------------------------------------|
-///  |       u16      |      u16       |  var        | var  |    u32    |  var  |
-///  |----------------|----------------|-------------|------|-----------|-------|
-///  | key_prefix_len | key_suffix_len |  key_suffix | meta | value_len | value |
-///  |--------------------------------------------------------------------------|
+///  |---------------------------------------------------------------------------------------------------------------------|
+///  |       u16      |      u16       |  var        | u64     | u8        | i64       | i64       | u32       |  var      |
+///  |----------------|----------------|-------------|---------|-----------|-----------|-----------|-----------|-----------|
+///  | key_prefix_len | key_suffix_len |  key_suffix | seq     | flags     | expire_ts | create_ts | value_len | value     |
```

Comment:
nit: let's keep `meta` and then have a separate line to describe the meta encoding (since expire_ts/create_ts are both optional fields now)

Reply:
i feel naming `meta` is fine here, but i was afraid that the term `meta` might conflict with the `meta` field used in other lsm engine's key encoding scheme like leveldb/rocksdb. in their context, `meta` is a `u64` field with a 7-byte `sequence number` and a 1-byte `key type`.

i suppose flattening these fields might help us reduce a bit burden on naming things (the hardest problem in computer science 🥲), and the fields in the key encoding scheme is not likely to extend more fields as other production lsm engines shows. wdyt?

Reply:
sgtm